lib/strings.nix

```diff
@@ -158,6 +158,24 @@ rec {
   */
   replicate = n: s: concatStrings (lib.lists.replicate n s);
 
+  /*
+    Remove leading and trailing whitespace from a string.
+
+    Whitespace is defined as any characters matching the regex
+    `[[:space:]]`.
```

Comment:
It would be good to be more specific here. You could mention that it uses [`builtins.match`](https://nix.dev/manual/nix/2.18/language/builtins.html?highlight=builtins.match#builtins-match) underneath, though digging down into what it uses for `[:space:]` makes me believe that it might be dependent on the `LC_CTYPE` env var, which would be bad..

Reply:
Are you suggesting we use a more explicit character class, along the lines of `[ \xa0\r\n]`?

Reply:
That would definitely avoid the ambiguity! I'm also wondering if non-Ascii spaces would even get matched by Nix

Reply:
Not sure how thorough we're looking to be here, but there's also [these space characters](https://en.wikipedia.org/wiki/Whitespace_character#Unicode)

Reply:
> there's also [these space characters](https://en.wikipedia.org/wiki/Whitespace_character#Unicode)

That's why I'd prefer we use a well defined class like `[:space:]`, but if it is non-deterministic that's less than ideal.

Either way, I'll add some more tests for whatever chars we deem to be whitespace/non-whitespace.

Reply:
> That's why I'd prefer we use a well defined class like `[:space:]`

It's [not well-defined](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html#tag_09_04
), that's precisely the issue ;)

> A character class expression shall represent the union of two sets:
>
> * The set of single characters that belong to the character class, as defined in the `LC_CTYPE` category in the current locale.
>
> * An unspecified set of multi-character collating elements.

Reply:
> Perhaps `[ \t\r\n]`?

This is done. @infinisil Let me know if you'd prefer a different approach.